src/Widgets/DisplayWidget.vala

```diff
@@ -21,7 +21,8 @@
 
 public class Display.DisplayWidget : Gtk.EventBox {
     public signal void set_as_primary ();
-    public signal void move_display (int delta_x, int delta_y);
+    public signal void move_display (double diff_x, double diff_y, Gdk.EventMotion event);
```

Comment:
You should not need to pass event as a parameter - the handler can access the event and the event state using `Gtk.get_current_event ()` and `Gtk.get_current_event_state (out state)`